### PR TITLE
Backend tests for default credentials

### PR DESCRIPTION
**Pull request by mjgui**
Closes [#1167]

## Tests

- Add tests for SMS and Telegram channel to ensure
  - only demo campaigns can get demo credentials
  - demo campaigns can only get demo credentials

### CODE REVIEW

File: .secrets.baseline [hunk 1]
```diff
@@ -1,9 +1,4 @@
 {
```
Comment (jeantanzj) [inline .secrets.baseline:1]: The changes in this file should have been in another PR. If you are changing something unrelated to the task, make another PR for it, so that it's easier to roll back.

But since this is a change that only affects development, it's ok.

**Review by jeantanzj — COMMENTED** [review]
Looks good so far

File: backend/src/telegram/routes/tests/telegram-campaign.routes.test.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    const demoCampaign = await createCampaign({ isDemo: true })
+
+    await request(app)
+      .post(`/campaign/${demoCampaign.id}/telegram/credentials`)
+      .send({
+        label: DefaultCredentialName.Telegram,
+      })
+
+    // Expect SecretManager to be called
+    expect(mockGetSecretValue).toHaveBeenCalled()
```
Comment (jeantanzj) [inline backend/src/telegram/routes/tests/telegram-campaign.routes.test.ts:101]: `toHaveBeenCalledWith`

File: .secrets.baseline (hunk 1 above)
Comment (mjgui) [reply .secrets.baseline:1]: Oh actually I tried to but I can't commit my changes without adding this file
(I think it might be due to the precommit hook for `detect-secrets`?)

Comment (lamkeewei) [reply .secrets.baseline:1]: Could be because of the version mismatch between your local version (1.1.0) of `detect-secrets` and the version that `.secrets.baseline` was generated with (0.14.3).

File: backend/src/sms/routes/tests/sms-campaign.routes.test.ts [hunk 3]
```diff
@@ -32,7 +26,119 @@ afterAll(async () => {
...
+
+  beforeAll(async () => {
+    const campaign = await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: ChannelType.SMS,
+      valid: false,
+    })
+    campaignId = campaign.id
+  })
```
Comment (mjgui) [inline backend/src/sms/routes/tests/sms-campaign.routes.test.ts:140]: Moving this down from above so it's clearer that it's used for template tests only

**Review by lamkeewei — APPROVED** [review]
lgtm! Squash and merge after resolving conflicts.

**Comment by mjgui** [comment]
Closing this. Will be done with PR #1204
